src/core_plugins/kibana/public/management/react/index-pattern/index-pattern-create/components/index-pattern-search/index-pattern-search.component.js

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import {
+  KuiFormRow,
+  KuiIcon,
+  KuiFlexGroup,
+  KuiFlexItem,
+} from 'ui_framework/components';
+
+import { InputPatternInputField } from './lib/index-pattern-input-field';
+
+const IndexPatternSearch = ({
+  hasExactMatches,
+  fetchIndices,
+}) => {
+  return (
+    <KuiFormRow
+      helpText="Patterns allow you to define dynamic index names using * as a wildcard"
+    >
+      <KuiFlexGroup>
+        <KuiFlexItem>
+          <InputPatternInputField
+            placeholder="Please enter..."
+            onChange={fetchIndices}
```

Comment:
This needs to debounce, it's fetching on every click.

Not sure this makes sense in this context, but in general: let's say one request is a tad slow and you keep typing. You might end up in a state where a _previous_ request end up responding after the next request. Basically:

```
fire off request 1
fire off request 2
receive response request 2
receive response request 1
```

You might need to have something that uniquely identifies each request to handle this (or some other way of rejecting the wrong state to end up in the store)